src/main/java/blusunrize/immersiveengineering/client/models/smart/ConnModelReal.java

```diff
@@ -183,16 +183,22 @@ public boolean equals(Object obj)
 			if (!(obj instanceof ExtBlockstateAdapter))
 				return false;
 			ExtBlockstateAdapter o = (ExtBlockstateAdapter) obj;
-			for (IProperty i : state.getPropertyNames())
+			for (IProperty<?> i : state.getPropertyNames())
 			{
+				if (!o.state.getProperties().containsKey(i))
+					return false;
 				Object valOther = o.state.getValue(i);
-				if (valOther == null || !valOther.equals(state.getValue(i)))
+				Object valMine = o.state.getValue(i);
```

Comment:
You're right, didn't see that.
I won't be anywhere near my computer for the next 1-2 weeks, so I can't fix this right now. If you want to merge this before I am back, please replace the `o.state.` by `state.` for `valMine` and replace the parameter to `equals` in the next line by `valMine`. Since I can't test the fix, I recommend that you check the cache size after the changes and check that rendering doesn't go crazy.